src/reducetensor.cpp

```diff
@@ -605,6 +622,8 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
 
         std::string param2 = param + " -DCK_PARAM_GRIDSIZE=" + std::to_string(gridSize_2) + " ";
 
+        std::string network_config2 = network_config + "_C2";
```

Comment:
@qianfengz Why this change (appending "_C2") is necessary?

Reply:
The second time and the first time use directly kernels since even though they are solving the same issue, so they should have different key for mapping in the kernel cache. 

One other thing that I want to mention is that, kernel cache is actually not queried,  since `Handle::AddKernel() ` actually only queries the Program cache,   and we does not use `Handle::HasKernel() + Handle::GetKernel()` here

Reply:
@qianfengz 
> The second time and the first time use different kernels even though they are for solving the same problem, so they should have different key for mapping in the kernel cache.

I do not understand, sorry.

Reply:
?